### PR TITLE
Amend/sidenav and title bar

### PR DESCRIPTION
**Pull request by M4R1KU**
This closes sadly nothing.

Summary of changes:
- ​removed old sidebar and replaced it with a new one
- added titlebar which will contain a title and if wanted a md-tab-group which manages child-routes
- grouped project-non-feature-components in new module
- added multiple themes and possibility to change between them (not completly final yet)
- fixed overlay styling until we update to the new @angular/material beta

I made sure to:
- [x] Add and / or update tests
- [x] Run the code formatter (`CTRL + ALT + L`)
- [x] Update any documentation (javadoc, domain model, erd, ...)
- [x] Follow the contribution guidelines
- [x] Update CI scripts on Semaphore (changes to `auth0.yml` or `application.yml`)

Please review @jmesserli / cc: @outcobra/developers.

### CODE REVIEW

**Comment by M4R1KU** [comment]
This PR is kind of blocking all other frontend task so please review it ASAP.

**Comment by jmesserli** [comment]
I don't see how a PR can block work, but I'll be looking at it today

**Comment by M4R1KU** [comment]
I neither do but thanks 😂

**Comment by jmesserli** [comment]
* Neither do I :trollface:

**Comment by jmesserli** [comment]
Currently looking at this.

**Comment by jmesserli** [comment]
The sidebar does not close when clicking on an item on mobile (at least in firefox responsive design mode)

**Comment by M4R1KU** [comment]
Which sidenav? the mobile?

Joel Messerli <notifications@github.com> schrieb am Di., 18. Apr. 2017,
19:54:

**Comment by jmesserli** [comment]
Yah, also you might want to look at this on github since I updated the comment :)

File: frontend/src/app/app.component.ts [hunk 1]
```diff
@@ -1,36 +1,63 @@
...
+    @HostBinding('class.oc-mobile')
     public mobile: boolean;
     public title = 'Outcobra';
 
+    public activeTheme: OCTheme;
+    public allThemes: Array<OCTheme> = OCTheme.values();
+
+    @ViewChild(MdSidenav) public sidenav: MdSidenav;
+
     public isEnglish: boolean = this.translateService.currentLang == 'en';
```
Comment (jmesserli) [inline frontend/src/app/app.component.ts]: Is this still needed?

**Comment by jmesserli** [comment]
|`frontend/src/app/oc-ui/components/oc-nav/oc-nav.component.scss:46`|
|---|
holy shit this nesting though :joy:

File: frontend/src/app/app.component.ts (hunk 1 above)
Comment (M4R1KU) [inline frontend/src/app/app.component.ts]: The language thingy?

Comment (jmesserli) [inline frontend/src/app/app.component.ts]: No the commented out constructor param

**Comment by M4R1KU** [comment]
You can ignore it this is uncommented on the frontend update branch bc(see
todos further down)

Joel Messerli <notifications@github.com> schrieb am Di., 18. Apr. 2017,
21:42:

**Comment by M4R1KU** [comment]
PR should be ready now.

File: frontend/src/app/app.component.ts [hunk 2]
```diff
@@ -37,12 +39,18 @@ export class AppComponent implements OnInit, AfterViewInit, AfterContentChecked
...
+                private router: Router,
                 /*private overlayContainer: OverlayContainer*/) {
     }
 
     ngOnInit() {
         this.recheckMobile();
         this.changeTheme(this.getThemeFromLocalStorage() || OCTheme.OCEAN);
+        this.router.events
+            .filter(event => event instanceof NavigationEnd)
+            .filter(() => isNotNull(this.sidenav))
```
Comment (jmesserli) [inline frontend/src/app/app.component.ts]: TBH I'm not sure if I like filter operations for something that does not interact with the Observable's elements, it seems like that would not be intended  (I'm sure it works though).
I would prefer an if statement in the last statement like this: 
```typescript
.do(() => 
    if (isNotNull(this.sidenav) && this.sidenav.opened) this.sidenav.close();
);
```
(I know I said I like it before but I changed my mind)

**Comment by M4R1KU** [comment]
Yeah it looks better so you can change it. I'm ok and I'm currently not
really able to change it myself

Joel Messerli <notifications@github.com> schrieb am Fr., 21. Apr. 2017,
08:12:

**Comment by jmesserli** [comment]
@M4R1KU Thanks for the reply, enjoy your vacation.

**Pull request merged**